src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -1293,5 +1295,14 @@ private bool IsPreviewWindow(ScriptFile scriptFile)
                 return false;
             }
         }
+
+        internal void DeletePeekDefinitionScripts()
+        {
+            // Delete temp folder created to store peek definition scripts
+            if (FileUtils.SafeDirectoryExists(FileUtils.PeekDefinitionTempFolder))
```

Comment:
As mentioned offline, let's ensure that each workspace has its own temp folder by creating a sub-folder for this the 1st time we need to save a file. Otherwise if you have 2 workspaces open and close one, the definitions files are deleted for the other workspace